packages/plugins/graphql/server/src/bootstrap.ts

```diff
@@ -101,39 +111,68 @@ export async function bootstrap({ strapi }: { strapi: Strapi }) {
     throw error;
   }
 
-  // Link the Apollo server & the Strapi app
+  // Create the route handlers for Strapi
+  const handler: Common.MiddlewareHandler[] = [];
+
+  // add cors middleware
+  if (cors) {
+    handler.push(cors());
+  }
+
+  // add koa bodyparser middleware
+  if (isObject(serverConfig.bodyParserConfig)) {
+    handler.push(bodyParser(serverConfig.bodyParserConfig));
+  } else if (serverConfig.bodyParserConfig) {
+    handler.push(bodyParser());
+  } else {
+    strapi.log.debug('Body parser has been disabled for Apollo server');
+  }
+
+  // add the Strapi auth middleware
+  handler.push((ctx: Context, next: Next) => {
+    ctx.state.route = {
+      info: {
+        // Indicate it's a content API route
+        type: 'content-api',
+      },
+    };
+
+    // allow graphql playground to load without authentication
+    // WARNING: this means graphql should not accept GET requests generally
+    // TODO: find a better way and remove this, it is causing issues such as https://github.com/strapi/strapi/issues/19073
+    if (ctx.request.method === 'GET') {
+      return next();
+    }
+
+    return strapi.auth.authenticate(ctx, next);
+  });
+
+  // add the graphql server for koa
+  handler.push(
+    koaMiddleware(server, {
+      // Initialize loaders for this request.
+      context: async ({ ctx }) => ({
+        state: ctx.state,
+        koaContext: ctx,
+      }),
+    })
+  );
+
+  // now that handlers are set up, add the graphql route to our apollo server
   strapi.server.routes([
     {
       method: 'ALL',
       path,
-      handler: [
-        (ctx, next) => {
-          ctx.state.route = {
-            info: {
-              // Indicate it's a content API route
-              type: 'content-api',
-            },
-          };
-
-          // allow graphql playground to load without authentication
-          if (ctx.request.method === 'GET') return next();
-
-          return strapi.auth.authenticate(ctx, next);
-        },
-
-        // Apollo Server
-        server.getMiddleware({
-          path,
-          cors: serverConfig.cors,
-          bodyParserConfig: serverConfig.bodyParserConfig,
-        }),
-      ],
+      handler,
       config: {
         auth: false,
       },
     },
   ]);
 
+  // add a method to check if playground was actually enabled to avoid re-implementing env and config checks
+  strapi.plugin('graphql').isPlaygroundEnabled = () => playgroundEnabled;
```

Comment:
Shouldn't we rename that to isLandingPageEnabled instead? (to anticipate the switch to the new option)

Reply:
Also, we could have implemented this method in a service instead (and reuse it above), it feels a bit weird to expose that at the plugin API level